signin_parser.rb

```diff
@@ -17,19 +17,26 @@ def initialize
     @browser = Capybara.current_session
   end
 
-  def sign_in(url = "", login = "", password = "", enter = "")
+  def sign_in(url = "", login = "", password = "", enter = "", quit = "")
     begin
       @browser.visit(url)
+    rescue Selenium::WebDriver::Error::InvalidArgumentError # URL error
+      return "URL #{url} is invalid"
     rescue Selenium::WebDriver::Error::UnknownError # Any visit error
-      return false
+      return "Visit error for #{url}"
     end
     @browser.fill_in('login', with: login)
     @browser.fill_in('password', with: password)
     @browser.click_button(enter)
+    begin
+      @browser.find('a', text: quit)
+    rescue Capybara::ElementNotFound
+      return "Login/password combination is invalid"
```

Comment:
I would raise specific class of error, let's say LoginFailed or UnableToLogin, something like that, returning string with message is not that useful when you developing method for other parts of code to use.